core/include/gnuradio-4.0/Graph.hpp

```diff
@@ -118,13 +121,15 @@ void forEachEdge(const GraphLike auto& root, Fn&& function, Edge::EdgeState filt
 
 } // namespace graph
 
-template<typename TSubGraph>
-class GraphWrapper : public BlockWrapper<TSubGraph> {
+template<typename TSelf, typename TSubGraph = TSelf>
+class GraphWrapper : public BlockWrapper<TSelf> {
+protected:
     std::unordered_multimap<std::string, std::string> _exportedInputPortsForBlock;
     std::unordered_multimap<std::string, std::string> _exportedOutputPortsForBlock;
 
 public:
-    GraphWrapper(gr::property_map init = {}) : gr::BlockWrapper<TSubGraph>(std::move(init)) {
```

Comment:
Please revert:
  * derived/user-defined blocks should not define their own constructor (disables the CRTP-ones).
  * if they defined one, it should be always `XXX(gr::property_map init = {})` to support non-C++ bindings and runtime polymorphism.

Reply:
Will do.

But mind that GraphWrapper is Graph's type-eraseure like BlockModel is for Block - it is *not* a gr::Block, no CRTP here.

Now that blocks only support property-map-based constructors, this should be possible to do instead of wrappers having to support an arbitrary number of construction arguments and perfect-forward them.